lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -815,13 +827,24 @@ static void optimizeBatchNorm(Function *F) {
       // Q = W * A
       // C = b * A + B
 
-      auto filterH = cast<Variable>(CV->getFilter())->getHandle<>();
-      auto cbiasH = cast<Variable>(CV->getBias())->getHandle<>();
+      Variable *filterV = getModifiableVariable(*CV->getFilter());
+      Variable *cbiasV = getModifiableVariable(*CV->getBias());
+
+      Variable *scaleV = getModifiableVariable(*BN->getScale());
+      Variable *biasV = getModifiableVariable(*BN->getBias());
+      Variable *meanV = getModifiableVariable(*BN->getMean());
+      Variable *var = getModifiableVariable(*BN->getVar());
```

Comment:
The BatchNorm variables don't technically have to be single-use, right?  Since we're actually modifying the Conv inputs?

Reply:
Good catch!